src/main/java/io/gingersnapproject/service/impl/DefaultDNSResolver.java

```diff
@@ -0,0 +1,29 @@
+package io.gingersnapproject.service.impl;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.gingersnapproject.service.DNSResolverService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class DefaultDNSResolver implements DNSResolverService {
```

Comment:
Do we need this to be it's own bean or can we just add the `resolve` code to `MembershipListener`?

Reply:
We can add the method. I don't think we'll need to resolve a DNS from somewhere else.